lib/injekt.rb

```diff
@@ -0,0 +1,47 @@
+class Array
+
+  def injekt(*args, &block)
+
+    block = parse_arguments(args, block, block_given?)
+    self.unshift(args.pop) unless args.empty?
+
+    do_injekt_iteration(self, block)
+    # do_injekt_recursion(self, block)
+
+  end
+
+  private
+
+  def parse_arguments(args, block, block_flag)
```

Comment:
Metrics/AbcSize: Assignment Branch Condition size for parse_arguments is too high. [15.17/15]